### PR TITLE
[DO NOT MERGE] WIP: Test hairpin fix for NLBs

### PR DESCRIPTION
**Pull request by gcs278**
Testing fix for https://issues.redhat.com/browse/OCPBUGS-9026

### CODE REVIEW

File: pkg/providers/v1/aws.go [hunk 1]
```diff
@@ -4523,6 +4523,19 @@ func v2toStatus(lb *elbv2.LoadBalancer) *v1.LoadBalancerStatus {
...
+		if err := wait.PollImmediate(10*time.Second, 5*time.Minute, func() (bool, error) {
+			addrs, err := net.LookupIP(ingress.Hostname)
+			if err != nil {
+				klog.Infof("waiting for loadbalancer domain %s to resolve...", ingress.Hostname)
+				return false, nil
+			}
+			ingress.IP = addrs[0].String()
+			return true, nil
+		}); err != nil {
+			klog.Fatalf("loadbalancer domain %s was unable to resolve:", ingress.Hostname)
```
Comment (Miciah) [inline pkg/providers/v1/aws.go:4537]: `klog.Fatal` seems extreme.  Is that just for the proof-of-concept?  

Would opportunistically setting `ingress.IP` be sufficient (i.e. set it if the lookup succeeds, but leave it unset if the lookup fails), or would that only make https://issues.redhat.com/browse/OCPBUGS-9026 less likely to occur and even more difficult to diagnose when it does occur?

Comment (Miciah) [reply pkg/providers/v1/aws.go:4537]: Hm, but I suppose you would want to keep retrying if the lookup initially failed, so we would need something to trigger a retry later on...

Comment (gcs278) [reply pkg/providers/v1/aws.go:4537]: Yea that could be changed, this is just a proof-of-concept.

I think the we still need to push on the AWS cloud team / SIG working group to see if they could provide the IP via their API.

Comment (elmiko) [reply pkg/providers/v1/aws.go:4537]: agreed about getting this from an official AWS API, `net.LookupIP` feels fragile to me.

**Comment by gcs278** [comment]
/close
https://issues.redhat.com/browse/OCPBUGS-9026 has been closed

**Pull request closed without merging**